third_party/move/move-compiler/src/inlining/visitor.rs

```diff
@@ -43,285 +43,6 @@ impl<'l, V: Visitor> Dispatcher<'l, V> {
-            | UnannotatedExp_::Constant(_, _)
-            | UnannotatedExp_::Break
-            | UnannotatedExp_::Continue
-            | UnannotatedExp_::UnresolvedError => {},
-        }
-    }
-
-    fn builtin_function(&mut self, fun: &mut BuiltinFunction) {
-        match &mut fun.value {
-            BuiltinFunction_::MoveTo(ty)
-            | BuiltinFunction_::MoveFrom(ty)
-            | BuiltinFunction_::BorrowGlobal(_, ty)
-            | BuiltinFunction_::Exists(ty)
-            | BuiltinFunction_::Freeze(ty) => self.type_(ty),
-            BuiltinFunction_::Assert(_) => {},
-        }
-    }
-
-    pub fn sequence(&mut self, seq: &mut Sequence) {
-        let mut scope_cnt = 0;
-        for item in seq.iter_mut() {
-            match &mut item.value {
-                SequenceItem_::Bind(decls, tys, e) => {
-                    self.exp(e.as_mut());
-                    self.types(tys.iter_mut().filter_map(|t| t.as_mut()));
-                    self.visitor.enter_scope();
-                    self.lvalue_list(decls, /*declared*/ true);
-                    scope_cnt += 1;
-                },
-                SequenceItem_::Declare(decls) => {
-                    self.visitor.enter_scope();
-                    self.lvalue_list(decls, /*declared*/ true);
-                    scope_cnt += 1;
-                },
-                SequenceItem_::Seq(e) => self.exp(e.as_mut()),
-            }
-        }
-        while scope_cnt > 0 {
-            self.visitor.exit_scope();
-            scope_cnt -= 1
-        }
-    }
-
-    pub fn lvalue_list(&mut self, decls: &mut LValueList, declared: bool) {
-        for lv in &mut decls.value {
-            self.lvalue(lv, declared)
-        }
-    }
-
-    fn lvalue(&mut self, lv: &mut LValue, declared: bool) {
-        match &mut lv.value {
-            LValue_::Var(var, ty) => {
-                self.type_(ty.as_mut());
-                if declared {
-                    self.visitor.var_decl(var)
-                } else {
-                    self.visitor.var_use(var)
-                }
-            },
-            LValue_::Unpack(_, _, tys, fields) | LValue_::BorrowUnpack(_, _, _, tys, fields) => {
-                self.types(tys.iter_mut());
-                for (_, _, (_, (ty, slv))) in fields.iter_mut() {
-                    self.type_(ty);
-                    self.lvalue(slv, declared);
-                }
-            },
-            LValue_::Ignore => {},
-        }
-    }
-}
-
-pub trait TypedVisitor {
-    fn exp(&mut self, _ex: &mut Exp) -> VisitorContinuation {
-        VisitorContinuation::Descend
-    }
-    fn ty(&mut self, _t: &mut Type) -> VisitorContinuation {
-        VisitorContinuation::Descend
-    }
-    fn enter_scope(&mut self) {}
-    fn var_decl(&mut self, _ty: &mut Type, _var: &mut Var) {}
-    fn var_use(&mut self, _ty: &mut Type, _var: &mut Var) {}
-    fn exit_scope(&mut self) {}
-}
-
-pub struct TypedDispatcher<'l, V: TypedVisitor> {
```

Comment:
Same for `TypedDispatcher`

Reply:
I removed TypedDispatcher and TypedVisitor, adding the missing type_ visiting functionality to Dispatcher/Visitor so it can be used for both cases.